x/simulation/params.go

```diff
@@ -188,3 +76,29 @@ func RandomParams(r *rand.Rand) Params {
 		BlockSizeTransitionMatrix: defaultBlockSizeTransitionMatrix,
 	}
 }
+
+//-----------------------------------------------------------------------------
+// Param change proposals
+
+// ParamChange defines the object used for simulating parameter change proposals
+type ParamChange struct {
+	Subspace string
+	Key      string
+	Subkey   string
+	SimValue func(r *rand.Rand) string
+}
+
+// NewSimParamChange creates a new ParamChange instance
+func NewSimParamChange(subspace, key, subkey string, simVal func(r *rand.Rand) string) ParamChange {
```

Comment:
For clarity, we've been typically trying to define function variable types outside of the function fields, something like:

```suggestion
var simValFn func(r *rand.Rand) string

func NewSimParamChange(subspace, key, subkey string, simVal simValFn) ParamChange {
```